tests/Common/RegexMatchHelperTests.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using System.Net;
+using System.Collections.Generic;
+using SNPN.Common;
+
+namespace SNPN.Test.Common
+{
+	public class RegexMatchHelperTests
+	{
+		[Fact]
+		void RegexMatchOnWordBoundaries()
+		{
+            Assert.True(RegexMatchHelper.MatchWholeWord("Hello there test.", "TEST"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Hello there,test how are you", "TEST"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Hello there,test,how are you", "test"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Test how are you", "test"));
+            Assert.True(RegexMatchHelper.MatchWholeWord(@"How are you
+                sure do hope you like multiline test
+                yep", "test"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Test, how are you", "test"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("-Test-!, how are you", "-test-"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Hello there test ", "test"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Will this annoy node", "node"));
+            Assert.True(RegexMatchHelper.MatchWholeWord("Will this annoy node?", "node"));
+
+            Assert.False(RegexMatchHelper.MatchWholeWord("Hello there testtest.", "test"));
+            Assert.False(RegexMatchHelper.MatchWholeWord("Should not match node-js?", "node"));
```

Comment:
If we use the regex I suggested, this will match. I'm okay with that. Seems better to be a bit noisier than miss things.